testsuite/tests/typing-misc/exp_denom.ml

```diff
@@ -36,8 +36,7 @@ type v = Cons
 Line 3, characters 18-22:
 3 | let _ = print_int Cons
                       ^^^^
-Error: The constructor "Cons" has type "v" but an expression was expected of type
-         "int"
+Error: This expression should not be a constructor, the expected type is "int"
```

Comment:
As above, there is less information than before.

Reply:
This can be fixed independently: the kind mismatch error messages was just not updated to use the improved handling of nominal expression that was added for type expression clashes.

Reply:
I think of doing that as a follow up PR.